graspy/align/seedless_procrustes.py

```diff
@@ -0,0 +1,373 @@
+
+    def _orthogonal_matrix_from_int(self, val_int, d):
+        val_bin = bin(val_int)[2:]
+        val_bin = "0" * (d - len(val_bin)) + val_bin
+        return np.diag(np.array([(float(i) - 0.5) * -2 for i in val_bin]))
+
+    def _optimal_transport(self, X, Y, Q):
+        # "E step" of the SeedlessProcrustes.
+        n, d = X.shape
+        m, _ = Y.shape
+        # initialize probability mass arrays & the cost matrix ; run sinkhorn
+        probability_mass_X = np.ones(n) / n
+        probability_mass_Y = np.ones(m) / m
+        cost_matrix = (
+            np.linalg.norm((X @ Q).reshape(n, 1, d) - Y.reshape(1, m, d), axis=2) ** 2
+        )
+        P = ot.sinkhorn(
+            a=probability_mass_X,
+            b=probability_mass_Y,
+            M=cost_matrix,
+            reg=self.optimal_transport_lambda,
+            numItermax=self.optimal_transport_eps,
+            stopThr=self.optimal_transport_eps,
+        )
+        return P
+
+    def _procrustes(self, X, Y, P_i):
+        # "M step" of the SeedlessProcurstes.
+        aligner = OrthogonalProcrustes()
+        Q = aligner.fit(X, P_i @ Y).Q_X
+        return Q
+
+    def _iterative_ot(self, X, Y, Q):
+        for i in range(self.iterative_num_reps):
+            P_i = self._optimal_transport(X, Y, Q)
+            Q = self._procrustes(X, Y, P_i)
+            c = np.linalg.norm(X @ Q - P_i @ Y, ord="fro")
+            if c < self.iterative_eps:
+                break
+        return P_i, Q
+
+    def fit(self, X, Y):
+        """
+        Uses the two datasets to learn matrices Q_X and Q_Y.
+        In seedless procrustes Q_X is a final solution of the to the iterative
+        optimal transport / procrustes algorithm and Q_Y is the identity
+        matrix.
+
+        Parameters
+        ----------
+        X: np.ndarray, shape (n, d)
+            First dataset of vectors. These vectors need to have same number of
+            dimensions as ones in Y, but the number of vectors can differ.
+
+        Y: np.ndarray, shape (m, d)
+            Second dataset of vectors. These vectors need to have same number
+            of dimensions as ones in X, but the number of vectors can differ.
+
+        Returns
+        -------
+        self: returns an instance of self
+        """
+
+        # check for numpy-ness, 2d-ness and finite-ness
+        if not isinstance(X, np.ndarray):
+            msg = f"first dataset is a {type(X)}, not an np.ndarray! "
+            raise TypeError(msg)
+        if not isinstance(Y, np.ndarray):
+            msg = f"first dataset is a {type(Y)}, not an np.ndarray! "
+            raise TypeError(msg)
+        X = check_array(X, accept_sparse=True, copy=True)
+        Y = check_array(Y, accept_sparse=True, copy=True)
+
+        # check for equal components and number of entries
+        if X.shape[1] != Y.shape[1]:
+            msg = "two datasets have different number of components!"
+            raise ValueError(msg)
+        _, d = X.shape
```

Comment:
refactor, same as `OrthogonalProcrustes`. Can have `BaseProcrustes` class if needed, or can be in `BaseAlign` depending on what we decide about `SignFlips`

Reply:
refactored; now a function in `BaseAlign`.